apps/src/blockly/addons/cdoFieldImageDropdown.js

```diff
@@ -109,6 +109,39 @@ export class CdoFieldImageDropdown extends FieldGridDropdown {
       this.dropdownDispose_.bind(this)
     );
   }
+
+  /**
+   * @override
+   * Ensure that the input value is a valid language-neutral option.
+   * The only change we made from the parent method is to not use the cache,
+   * as the list of options can change.
+   *
+   * @param newValue The input value.
+   * @returns A valid language-neutral option, or null if invalid.
+   */
+  doClassValidation_(newValue) {
+    /* Begin CDO Customization */
+    const options = this.getOptions(false); // false = do not use cache
```

Comment:
It's unfortunate that this boolean wasn't an option constructor argument because then I think we could potentially just call `super.doClassValidation_(false)`. That is to say, this is probably the best way to handle this short of requesting a change in Blockly core or the plugin.